service/nodeconfig/v1/resource/node/create.go

```diff
@@ -103,5 +104,20 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 		}
 	}
 
+	{
+		n := v1.NamespaceDefault
+
+		c := v1alpha1.NodeConfigStatusCondition{
+			Status: "True",
+			Type:   "Drained",
+		}
+		customObject.Status.Conditions = append(customObject.Status.Conditions, c)
+
+		_, err := r.g8sClient.CoreV1alpha1().NodeConfigs(n).Update(&customObject)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
```

Comment:
Updating the node config status after draining so the provider operators know the draining is done. The provider operators can then remove the custom object.

Reply:
Nice. <3